packages/keyring-api/src/api/v2/keyring-capabilities.ts

```diff
@@ -0,0 +1,87 @@
+import {
+  array,
+  boolean,
+  exactOptional,
+  nonempty,
+  object,
+  type Infer,
+} from '@metamask/superstruct';
+
+import {
+  ExportPrivateKeyFormatStruct,
+  ImportPrivateKeyFormatStruct,
+} from './private-key';
+import { CaipChainIdStruct } from '../caip';
+
+/**
+ * Struct for {@link KeyringCapabilities}.
+ */
+export const KeyringCapabilitiesStruct = object({
+  /**
+   * List of CAIP-2 chain IDs that this keyring supports.
+   */
+  scopes: nonempty(array(CaipChainIdStruct)),
+  /**
+   * BIP-44 capabilities supported by this keyring.
+   */
+  bip44: exactOptional(
+    object({
+      /**
+       * Whether the keyring supports deriving accounts from a specific BIP-44 path.
+       */
+      derivePath: boolean(),
+      /**
+       * Whether the keyring supports deriving accounts from a BIP-44 account index.
+       */
+      deriveIndex: boolean(),
+      /**
+       * Whether the keyring supports BIP-44 account discovery.
+       */
+      discover: boolean(),
+    }),
+  ),
+  /**
+   * Private key capabilities supported by this keyring.
+   */
+  privateKey: exactOptional(
+    object({
+      /**
+       * List of supported formats for importing private keys.
+       */
+      importFormats: array(ImportPrivateKeyFormatStruct),
+      /**
+       * List of supported formats for exporting private keys.
+       */
+      exportFormats: array(ExportPrivateKeyFormatStruct),
```

Comment:
Might have missed that during the ADR, but I now wonder if they should be also `exactOptional`?

Like we could combine `bip44` and `privateKey.exportFormats` and not have any way to import a private key? 🤔

Really not sure about this, just a thought. Since we can use `[]`, that could also work to know if import/export is supported (but that's less idiomatic and straightforward IMO)